pkg/ipam-node/cleaner/cleaner.go

```diff
@@ -84,13 +88,19 @@ func (c *cleaner) Start(ctx context.Context) {
 
 func (c *cleaner) loop(ctx context.Context) error {
 	logger := logr.FromContextOrDiscard(ctx)
-	store, err := c.store.Open(ctx)
+	session, err := c.store.Open(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to open store: %v", err)
 	}
 	allReservations := map[string]struct{}{}
-	for _, poolName := range store.ListPools() {
-		for _, reservation := range store.ListReservations(poolName) {
+	emptyPools := []string{}
+	for _, poolName := range session.ListPools() {
+		poolReservations := session.ListReservations(poolName)
+		if len(poolReservations) == 0 {
+			emptyPools = append(emptyPools, poolName)
+			continue
```

Comment:
nit: continue not needed, up to u if you think its safer to keep it here.

Reply:
I prefer to continue explicitly